jackson-jq/src/main/java/net/thisptr/jackson/jq/internal/tree/binaryop/BinaryOperatorExpression.java

```diff
@@ -42,62 +40,41 @@ public String toString() {
 	}
 
 	public enum Operator {
-		ASSIGN("=", 6, Associativity.RIGHT, Assignment.class),
-		UDPATE("|=", 6, Associativity.RIGHT, UpdateAssignment.class),
-		DEFAULT_EQUAL("//=", 6, Associativity.RIGHT, ComplexAlternativeAssignment.class),
-		PLUS_EQUAL("+=", 6, Associativity.RIGHT, ComplexPlusAssignment.class),
-		MINUS_EQUAL("-=", 6, Associativity.RIGHT, ComplexMinusAssignment.class),
-		TIMES_EQUAL("*=", 6, Associativity.RIGHT, ComplexMultiplyAssignment.class),
-		DIVIDE_EQUAL("/=", 6, Associativity.RIGHT, ComplexDivideAssignment.class),
-		MODULO_EQUAL("%=", 6, Associativity.RIGHT, ComplexModuloAssignment.class),
-		DEFAULT("//", 5, Associativity.LEFT, AlternativeOperatorExpression.class),
-		OR("or", 4, Associativity.LEFT, BooleanOrExpression.class),
-		AND("and", 4, Associativity.LEFT, BooleanAndExpression.class),
-		LESS_EQUAL("<=", 3, Associativity.LEFT, CompareLessEqualTest.class),
-		LESS("<", 3, Associativity.LEFT, CompareLessTest.class),
-		GREATER_EQUAL(">=", 3, Associativity.LEFT, CompareGreaterEqualTest.class),
-		GREATER(">", 3, Associativity.LEFT, CompareGreaterTest.class),
-		EQUAL("==", 3, Associativity.LEFT, CompareEqualTest.class),
-		NOT_EQUAL("!=", 3, Associativity.LEFT, CompareNotEqualTest.class),
-		PLUS("+", 2, Associativity.LEFT, PlusExpression.class),
-		MINUS("-", 2, Associativity.LEFT, MinusExpression.class),
-		MODULO("%", 1, Associativity.LEFT, ModuloExpression.class),
-		DIVIDE("/", 1, Associativity.LEFT, DivideExpression.class),
-		TIMES("*", 1, Associativity.LEFT, MultiplyExpression.class);
+		ASSIGN("=", 6, Associativity.RIGHT),
```

Comment:
Can you change this to use a factory lambda instead, in order to avoid the lengthy switch-case later? E.g.

```java
    ASSIGN("=", 6, Associativity.RIGHT, (lhs, rhs, version) -> new Assignment(lhs, rhs)),
```

Reply:
So I was trying this and this was looking a little bit more complex to understand as the number of characters in a line was exceeding 100 characters.

Instead I just tried to use abstract method, which makes more sense and is a much cleaner approach for future use-cases too, if any?